frontend/src/utils/context.tsx

```diff
@@ -0,0 +1,16 @@
+import * as React from 'react';
+
+// Define the type so that createContext understands what to expect for focusedUser
+type AppContextType = {
+    focusedUser: [number, React.Dispatch<React.SetStateAction<number>>];
+};
+
+// Even though this is initialized as null, using it will supply the correct type
+export const AppContext = React.createContext<AppContextType>({
+    focusedUser: null,
+});
+
+export default function useFocusedUser() {
+    const context = React.useContext(AppContext);
+    return context.focusedUser;
+}

```

Comment:
Ok there's a lot going on here.
At the base level, I create a context called `AppContext`. By using `<AppContext.Provider value={something}>` I initialize the context, which is serving as our global state.

Then I define a custom hook, `useFocusedUser` to let us use the context without having to pass it down the entire component hierarchy as props.

Lastly, at the top I explicitly define the type of the context. Without this, it will be of type "any". Now it's not.